src/librustc_mir/transform/const_prop.rs

```diff
@@ -790,11 +791,11 @@ impl<'tcx> Visitor<'tcx> for CanConstProp {
             // FIXME(oli-obk): we could be more powerful here, if the multiple writes
             // only occur in independent execution paths
             MutatingUse(MutatingUseContext::Store) => {
-                if self.found_assignment[local] {
+                if self.found_assignment.contains(local) {
                     trace!("local {:?} can't be propagated because of multiple assignments", local);
                     self.can_const_prop[local] = ConstPropMode::NoPropagation;
                 } else {
-                    self.found_assignment[local] = true
+                    self.found_assignment.insert(local);
```

Comment:
FWIW we should be just `insert(local)` above I think, and then checking if we changed, rather than checking and then inserting. Should be a bit faster, though possibly what this lowers to anyway :)

Reply:
Yep, that totally makes sense!